File: VB/WindowsApplication3/Data/Record.vb
Imports System.ComponentModel
Imports System.Drawing

Namespace DXSample.Data

    Public Class Record

        Public Sub New()
        End Sub

        Public Property ID As Integer

        Public Property ParentID As Integer?

        Public Property Text As String

        Public Property Info As String

        Public Property Value As Double

        Public Property Dt As Date

        Public Property State As Boolean

        Public Property Image As Image

        Public Property Details As BindingList(Of Detail)

        Public Overrides Function ToString() As String
            Return String.Format("ID = {0}, Text = {1}", ID, Text)
        End Function

        Public Property Detail As Detail
    End Class

    Public Class Detail

        Public Sub New()
            Details = New BindingList(Of Detail)()
        End Sub

        Public Property ID As Integer

        Public Property Text As String

        Public Property Info As String

        Public Property Details As BindingList(Of Detail)

        Public Overrides Function ToString() As String
            Return Text
        End Function
    End Class

    Public Class DataHelper

        Public Shared Function GetData(ByVal count As Integer, ByVal detailCount As Integer) As BindingList(Of Record)
            Dim records As BindingList(Of Record) = New BindingList(Of Record)()
            Dim details As BindingList(Of Detail) = New BindingList(Of Detail)()
            Dim testDetails As BindingList(Of Detail) = New BindingList(Of Detail)()
            For j As Integer = 0 To detailCount - 1
                details.Add(New Detail() With {.ID = j, .Text = String.Format("Text {0}", j), .Info = String.Format("Info {0}", j)})
                testDetails.Add(New Detail() With {.ID = j, .Text = String.Format("Test Text {0}", j), .Info = String.Format(" Test Info {0}", j)})
            Next

            For i As Integer = 0 To count - 1
                records.Add(New Record() With {.ParentID = -1, .ID = i, .Text = String.Format("Text {0}", i), .Info = String.Format("Info {0}", i), .Dt = Date.Now.AddDays(i), .State = i Mod 2 = 0, .Image = SystemIcons.Information.ToBitmap(), .Details = New BindingList(Of Detail)(details), .Value = i * 0.00001, .Detail = New Detail() With {.ID = i, .Text = String.Format("Text {0}", i)}})
            Next

            Return records
        End Function

        Public Shared Function GetData(ByVal count As Integer) As BindingList(Of Record)
            Dim records As BindingList(Of Record) = New BindingList(Of Record)()
            For i As Integer = 0 To count - 1
                records.Add(New Record() With {.ID = i, .ParentID = i Mod 5, .Text = String.Format("Text {0}", i), .Info = String.Format("Info {0}", i), .Dt = Date.Now.AddDays(i), .Value = 10.1 * i, .State = False, .Image = SystemIcons.Information.ToBitmap(), .Detail = New Detail() With {.ID = i}})
            Next

            Return records
        End Function

        Public Shared Function GetDetailData(ByVal count As Integer) As BindingList(Of Detail)
            Dim records As BindingList(Of Detail) = New BindingList(Of Detail)()
            For i As Integer = 0 To count - 1
                records.Add(New Detail() With {.ID = i, .Text = String.Format("Text {0}", i), .Info = String.Format("Info {0}", i)})
            Next

            Return records
        End Function
    End Class
End Namespace

File: CS/WindowsApplication3/Data/Record.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Linq;

namespace DXSample.Data {
    public class Record {
        public Record() {
        }
        public int ID { get; set; }
        public int? ParentID { get; set; }
        public string Text { get; set; }
        public string Info { get; set; }
        public double Value { get; set; }
        public DateTime Dt { get; set; }
        public bool State { get; set; }
        public Image Image { get; set; }
        public BindingList<Detail> Details { get; set; }
        public override string ToString() {
            return string.Format("ID = {0}, Text = {1}", ID, Text);
        }
        public Detail Detail { get; set; }

    }
    public class Detail {
        public Detail() {
            Details = new BindingList<Detail>();
        }
        public int ID { get; set; }
        public string Text { get; set; }
        public string Info { get; set; }
        public BindingList<Detail> Details { get; set; }
        public override string ToString() {
            return Text;
        }
    }

    public class DataHelper {
        public static BindingList<Record> GetData(int count, int detailCount){
            BindingList<Record> records = new BindingList<Record>();
            BindingList<Detail> details = new BindingList<Detail>();
            BindingList<Detail> testDetails = new BindingList<Detail>();
            for (int j = 0; j < detailCount; j++) {
                details.Add(new Detail() {
                    ID = j,
                    Text = string.Format("Text {0}", j),
                    Info = string.Format("Info {0}", j)
                });
                testDetails.Add(new Detail() {
                    ID = j,
                    Text = string.Format("Test Text {0}", j),
                    Info = string.Format(" Test Info {0}", j)
                });
            }
            for (int i = 0; i < count; i++)
                records.Add(new Record() {
                    ParentID = -1,
                    ID = i,
                    Text = string.Format("Text {0}", i),
                    Info = string.Format("Info {0}", i),
                    Dt = DateTime.Now.AddDays(i),
                    State = i % 2 == 0,
                    Image = SystemIcons.Information.ToBitmap(),
                    Details = new BindingList<Detail>(details),
                    Value = i * 0.00001,
                    Detail = new Detail() { ID = i, Text = string.Format("Text {0}", i) }
                });

            return records;
        }
        public static BindingList<Record> GetData(int count) {
            BindingList<Record> records = new BindingList<Record>();
            for (int i = 0; i < count; i++)
                records.Add(new Record() {
                    ID = i,
                    ParentID = i % 5,
                    Text = string.Format("Text {0}", i),
                    Info = string.Format("Info {0}", i),
                    Dt =  DateTime.Now.AddDays(i),
                    Value = 10.1 * i,
                    State = false,
                    Image = SystemIcons.Information.ToBitmap(),
                    Detail = new Detail() { ID = i }
                });
            return records;
        }
        public static BindingList<Detail> GetDetailData(int count) {
            BindingList<Detail> records = new BindingList<Detail>();
            for (int i = 0; i < count; i++)
                records.Add(new Detail() {
                    ID = i,
                    Text = string.Format("Text {0}", i),
                    Info = string.Format("Info {0}", i)
                });
            return records;
        }
    }
}
